### PR TITLE
Close loop - Ansible job invocation shows wrong info after remote execution job

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Execution of Ansible job on multiple hosts simultaneously

**Test result:**

```
pytest tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_multiple_host
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.11.1, pytest-7.3.1, pluggy-1.0.0
Mandatory Requirements Available: jinja2==3.1.2
Optional Requirements Available: sphinx==7.0.1 manage>=0.1.13
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo
configfile: pyproject.toml
plugins: services-2.2.1, mock-3.10.0, reportportal-5.1.7, ibutsu-2.2.4, xdist-3.2.1, cov-3.0.0
collected 8 items / 7 deselected / 1 selected                                                                                                                                                                     

tests/foreman/api/test_ansible.py .                                                                                                                                                                         [100%]
============================================================================ 1 passed, 7 deselected, 2 warnings in 1714.08s (0:28:34) =============================================================================
```

### CODE REVIEW

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:167]: can we use API endpoints for assigning and listing ansible roles assigned?

File: tests/foreman/api/test_ansible.py
Comment (shubhamsg199) [inline tests/foreman/api/test_ansible.py:162]:
```suggestion
    for host in hosts:
        host.install_katello_ca(target_sat)
        host.register_contenthost(module_org.label, force=True)
        assert host.subscribed
        host.add_rex_key(satellite=target_sat)
        id = target_sat.nailgun_smart_proxy.id
        target_host = host.nailgun_host
```

File: tests/foreman/api/test_ansible.py
Comment (shubhamsg199) [inline tests/foreman/api/test_ansible.py:163]: should we check some status after syncing?

**Review by ogajduse — COMMENTED** [review]
@adarshdubey-star could you try to implement the `multi_manager` from  SatelliteQE/broker#215 in this test/fixture? You could create a fixture that yields that many hosts as we have defined in `supportability.yaml`. I can also imagine that this fixture could be parametrizable by `rhel_ver_match` and `rhel_ver_list` markers.
I find that a better solution than juggling with existing fixtures.

File: tests/foreman/api/test_ansible.py
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py:167]: I believe for listing we are using the API endpoint only and for assigning I wanted to do it via the name of the role,
else I have to fetch the ansible_role_id explicitly again as previously we were doing it by hardcoding the id as 1, which was not working properly when the roles were re-assigned in automation.

File: tests/foreman/api/test_ansible.py
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py:163]: post syncing there is an assertion of the role which got assigned I believe that is a good check we are doing whether the sync went well, it is a basic import of the role.

**Comment by adarshdubey-star** [comment]
> @adarshdubey-star could you try to implement the `multi_manager` from [SatelliteQE/broker#215](https://github.com/SatelliteQE/broker/pull/215) in this test/fixture? You could create a fixture that yields that many hosts as we have defined in `supportability.yaml`. I can also imagine that this fixture could be parametrizable by the `rhel_ver_match` and `rhel_ver_list` markers. I find that a better solution than juggling with existing fixtures.

Sure, I can try that post merging thanks @ogajduse.
I was looking for something like this,  also does this functionality helps to spin capsule /sat/rhel in one go? just curious 
cc: @JacobCallahan

**Comment by ogajduse** [comment]
> > @adarshdubey-star could you try to implement the `multi_manager` from [SatelliteQE/broker#215](https://github.com/SatelliteQE/broker/pull/215) in this test/fixture? You could create a fixture that yields that many hosts as we have defined in `supportability.yaml`. I can also imagine that this fixture could be parametrizable by the `rhel_ver_match` and `rhel_ver_list` markers. I find that a better solution than juggling with existing fixtures.
> 
> Sure, I can try that post merging thanks @ogajduse. I was looking for something like this, also does this functionality helps to spin capsule /sat/rhel in one go? just curious cc: @JacobCallahan

I would hold off on merging and I would prefer the new broker feature to be used in this PR before we merge it.
However, if this test is critical to the product work, let's do it the way you suggest @adarshdubey-star.

**Comment by adarshdubey-star** [comment]
> > > @adarshdubey-star could you try to implement the `multi_manager` from [SatelliteQE/broker#215](https://github.com/SatelliteQE/broker/pull/215) in this test/fixture? You could create a fixture that yields that many hosts as we have defined in `supportability.yaml`. I can also imagine that this fixture could be parametrizable by the `rhel_ver_match` and `rhel_ver_list` markers. I find that a better solution than juggling with existing fixtures.
> > 
> > 
> > Sure, I can try that post merging thanks @ogajduse. I was looking for something like this, also does this functionality helps to spin capsule /sat/rhel in one go? just curious cc: @JacobCallahan
> 
> I would hold off on merging and I would prefer the new broker feature to be used in this PR before we merge it. However, if this test is critical to the product work, let's do it the way you suggest @adarshdubey-star.

Hey, @ogajduse there is an on_qa BZ that I want to verify with this automation, I will later try the multimanager setup.

File: tests/foreman/api/test_ansible.py
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py]: updated

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_multiple_host

File: pytest_fixtures/core/contenthosts.py
Comment (ogajduse) [inline pytest_fixtures/core/contenthosts.py:94]:
```suggestion
    """A fixture that provides a rhel9 content host object"""
```

File: tests/foreman/api/test_ansible.py
Comment (ogajduse) [inline tests/foreman/api/test_ansible.py:157]: Could you use the Global Registration instead of the old registration way? It is deprecated already. See [this chat discussion](https://redhat-internal.slack.com/archives/C04T3JZ5W8K/p1685545990914279)

**Review by ogajduse — CHANGES_REQUESTED** [review]

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_multiple_host

File: tests/foreman/api/test_ansible.py
Comment (ogajduse) [inline tests/foreman/api/test_ansible.py:169]: I would suggest against crafting the command in the test. Use the existing method instead.

What is the reason for forcing the registration?

```suggestion
       host.register(module_org, module_location, module_ak_with_synced_repo.name, target_sat)
       assert host.registered
```

**Review by ogajduse — CHANGES_REQUESTED** [review]

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_multiple_host

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_multiple_host

**Review by shubhamsg199 — APPROVED** [review]
Ack

**Review by Gauravtalreja1 — APPROVED** [review]

File: tests/foreman/api/test_ansible.py
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py]: Deleting the chat link resolved the comment.

**Pull request merged**
